flink-runtime/src/test/java/org/apache/flink/streaming/api/operators/SourceOperatorTest.java

```diff
@@ -273,6 +273,20 @@ public void testMetricGroupTracksSplitWatermark() throws Exception {
                 expectedWatermark);
     }
 
+    @Test
+    public void testMetricGroupReturnsDefaultIfNoSplitWatermark() throws Exception {
```

Comment:
nit: can we add a test where some of the splits have WMs and some do not?

Reply:
Yes. Im not sure what usecase you are trying to cover here though 🤔

Reply:
Agreed - it just seemed like a permutation to check. Thanks for adding anyway.